lib/env.ts

```diff
@@ -0,0 +1,65 @@
+import { Option, fromNullable } from "fp-ts/lib/Option";
+import * as t from "io-ts";
+
+export const EnvKey = t.union([
+  t.literal("SPID_SESSION_TOKEN"),
+  t.literal("IO_BACKEND_HOST"),
+  t.literal("IO_BACKEND_BASEPATH")
+]);
+
+export type EnvKey = t.TypeOf<typeof EnvKey>;
+
+export const EnvMap = t.record(EnvKey, t.string);
+export type EnvMap = Partial<t.TypeOf<typeof EnvMap>>;
+
+const pick = (obj: any) =>
+  Object.keys(obj).reduce<EnvMap>(
+    (p, k: string) =>
+      EnvKey.decode(k).fold(
+        () => p,
+        (key: EnvKey) => ({ ...p, [key]: obj[key] })
+      ),
+    {}
+  );
+
+class Env {
+  private env: EnvMap;
+  private constructor(args: EnvMap) {
+    const penv = pick(process.env);
+    this.env = { ...penv, ...args };
+  }
+
+  static instance() {
+    return new Env({});
+  }
+
+  get(key: EnvKey): Option<string> {
```

Comment:
avoid using getter and setters if not absolutely necessary, they hide what's happening